packages/doctype-tile/src/__tests__/tile-doctype.test.ts

```diff
@@ -267,6 +267,41 @@ describe('TileDoctypeHandler', () => {
     expect(state).toMatchSnapshot()
   })
 
+  it.only('squashes record correctly', async () => {
+    const deepCopy = o => DoctypeUtils.deserializeState(DoctypeUtils.serializeState(o))
+    const tileDoctypeHandler = new TileDoctypeHandler()
+
+    const genesisRecord = await TileDoctype.makeGenesis({ content: { test: 'data' }, metadata: { controllers: [did.id] } }, context)
+    await context.ipfs.dag.put(genesisRecord, FAKE_CID_1)
+    const payload = dagCBOR.util.deserialize(genesisRecord.linkedBlock)
+    await context.ipfs.dag.put(payload, genesisRecord.jws.link)
+    // apply genesis
+    const genesisState = await tileDoctypeHandler.applyRecord(genesisRecord.jws, FAKE_CID_1, context)
+
+    // make a first update
+    let doctype = new TileDoctype(genesisState, context)
+    const signedRecord1 = await TileDoctype._makeRecord(doctype, did, { other: { obj: 'content' } })
+
+    await context.ipfs.dag.put(signedRecord1, FAKE_CID_2)
+    const sPayload1 = dagCBOR.util.deserialize(signedRecord1.linkedBlock)
+    await context.ipfs.dag.put(sPayload1, signedRecord1.jws.link)
+    // apply signed
+    const state1 = await tileDoctypeHandler.applyRecord(signedRecord1.jws, FAKE_CID_2, context, deepCopy(genesisState))
+
+    // make a second update that squashes the first
+    doctype = new TileDoctype(state1, context)
+    const signedRecord2 = await TileDoctype._makeRecord(doctype, did, { other: { obj2: 'fefe' } })
+
+    await context.ipfs.dag.put(signedRecord2, FAKE_CID_3)
+    const sPayload2 = dagCBOR.util.deserialize(signedRecord2.linkedBlock)
+    await context.ipfs.dag.put(sPayload2, signedRecord2.jws.link)
+
+    // apply signed
+    const state2 = await tileDoctypeHandler.applyRecord(signedRecord2.jws, FAKE_CID_3, context, deepCopy(genesisState))
+
+    expect(state2).toMatchSnapshot()
```

Comment:
sigh, I get why the snapshots are useful and convenient, but I gotta admit I'm not a huge fan.  It means that it's totally unclear in the actual test what the desired behavior really is.  What goes wrong without your fix?  Is the content of the document wrong, or just some of the log structure is less ideal in some way?  Since I thought this bug was a blocker to something else, I assume that there's a real behavior change that's observable to the end user in some way?  Can we actually test that behavior explicitly here?  Can be in addition to the snapshot check, which is still useful for at least noticing when data formats change, but doesn't actually tell us if the change is good or bad.

Reply:
The test in this case crashes without the fix so wasn't exactly clear how to show this here. Without the fix, line 300 would throw an error.

Reply:
Oh interesting.  So was squashing just totally broken and totally untested?

Reply:
I think it worked in some cases but not all.